avm/res/compute/virtual-machine-scale-set/main.bicep

```diff
@@ -85,9 +85,8 @@ param licenseType string = ''
 param extensionDomainJoinPassword string = ''
 
 @description('Optional. The configuration for the [Domain Join] extension. Must at least contain the ["enabled": true] property to be executed.')
-param extensionDomainJoinConfig object = {
-  enabled: false
-}
+@secure()
+param extensionDomainJoinConfig object = {}
```

Comment:
Why changing the logic here and in the condition. It makes the module less consistent and hence harder to maintain, given that all other extensions use the block in its original way. Did something break?

Reply:
There was a security issue in the VM module, that mandated the extensionDomainJoinConfig object to be secure. Same applies for VMSS.

Reply:
if it is secure, Bicep does not allow clear text properties inside the object

Reply:
@rahalan - you break my heart